script.js

```diff
@@ -1,4 +1,68 @@
+// initialise first mode asking input
+var mode = 'need input';
+
+// two dice array for ramdom dice roll
+var diceOne = [];
+var diceTwo = [];
+
+// player 1 and 2 number
+var player1Dice = [];
+var player2Dice = [];
+
+// random dice generator for two dice
+var generateDiceNum = function () {
+  return Math.ceil(Math.random() * 6);
+};
+
+// function to concat first dice number
+function concatDiceNum1() {
+  return (diceOne[diceOne.length - 1]).toString() + (diceTwo[diceTwo.length - 1]).toString();
+}
+// function to concat second dice number
+function concatDiceNum2() {
+  return (diceTwo[diceTwo.length - 1]).toString() + (diceOne[diceOne.length - 1]).toString();
+}
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // if input is empty, goto need input mode
+  if (input == '') {
+    mode = 'need input';
+    return 'To play the game: input "p1" for player 1 to roll or "p2" for player 2 to roll' + '<br>' + 'you can change to player 2 after player 1 picks the dice number.';
+  }
+  // if input is "p1" goto player1 mode
+  if (input == 'p1') {
+    mode = 'player1';
+    diceOne.push(generateDiceNum());
+    diceTwo.push(generateDiceNum());
+    return 'Player 1' + '<br><br>' + 'Dice 1: You rolled ' + diceOne[diceOne.length - 1] + ' 🎲' + '<br><br>' + 'Dice 2: You rolled ' + diceTwo[diceTwo.length - 1] + ' 🎲' + '<br><br>' + 'Type either "dice1" or "dice2" to pick your first numeral of the combined number';
```

Comment:
Is there any way we can abstract this message generation into a helper function so that we don't have to keep repeating this string in the `main` function?